frontend/src/services/notebookService.ts

```diff
@@ -290,24 +291,35 @@ export const startNotebook = (data: StartNotebookData): Promise<Notebook> => {
   });
 };
 
-export const stopNotebook = (projectName: string, notebookName: string): Promise<Notebook> => {
+export const stopNotebook = async (
+  projectName: string,
+  notebookName: string,
+): Promise<Notebook> => {
   const dateStr = new Date().toISOString().replace(/\.\d{3}Z/i, 'Z');
-  const patch: RecursivePartial<Notebook> = {
+  const notebook = await getNotebook(projectName, notebookName);
+  const patch: RecursivePartial<Notebook> = _.merge({}, notebook, {
     metadata: { annotations: { 'kubeflow-resource-stopped': dateStr } },
-  };
-
-  return patchNotebook(projectName, notebookName, patch);
+  });
+  const url = `/api/notebooks/${projectName}/${notebookName}`;
+  return axios.patch(url, patch).then((response) => {
+    return response.data;
+  });
 };
 
-export const patchNotebook = (
+export const replaceNotebook = async (
```

Comment:
This code is not the best -- but it seems the code surrounding you is not that great either.

We fetch the notebook a few times over when we have access to the latest notebook in the Context of the NotebookController... we need to probably fix all of this, because it seems we are just compounding issue on top of issue.

I think we also want to rewrite the API to manage assembling of the Notebook server side -- for security reasons.

All of this is going to be rewritten unfortunately.